### PR TITLE
uri parser

### PR DESCRIPTION
**Pull request by earonesty**

### CODE REVIEW

File: notanorm/base.py [hunk 1]
```diff
@@ -187,6 +191,19 @@ def __init__(self, *args, **kws):
...
+        if cls.uri_name:
+            cls.__known_drivers[cls.uri_name] = cls
+        cls.__known_drivers[cls.__name__] = cls
+
+    @classmethod
+    def get_driver_by_name(cls, name) -> Type["DbBase"]:
+        return cls.__known_drivers.get(name)
+
+    @classmethod
+    def uri_adjust(cls, args: List, kws: Dict):
```
Comment (higleyc) [inline notanorm/base.py:204]: would be nice for this to have a docstring, mentioning the arg modification

File: tests/test_notanorm.py [hunk 2]
```diff
@@ -915,3 +929,38 @@ def test_mysql_op_error(db):
...
+    assert kws == {"timeout": 5.1}
+
+    typ, args, kws = parse_db_uri("mysql:host=localhost,port=45")
+    from notanorm import MySqlDb
+
+    assert typ == MySqlDb
+    assert kws == {"host": "localhost", "port": 45}
+
+    typ, args, kws = parse_db_uri("mysql:localhost,port=45")
+    from notanorm import MySqlDb
```
Comment (higleyc) [inline tests/test_notanorm.py:952]: multiply imported

File: notanorm/base.py (hunk 1 above)
Comment (earonesty) [reply notanorm/base.py:204]: k done

File: tests/test_notanorm.py (hunk 2 above)
Comment (earonesty) [reply tests/test_notanorm.py]: thanks

**Review by higleyc — APPROVED** [review]

**Pull request merged**